consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java

```diff
@@ -370,6 +365,21 @@ private String calculateSnapshotPath(String snapshotId, String originalFilePath)
     return originalFilePath.substring(originalFilePath.indexOf(snapshotId));
   }
 
+  private long getLatestSnapshotIndex() {
+    long snapShotIndex = 0;
+    File directory = new File(storageDir);
+    File[] versionFiles = directory.listFiles((dir, name) -> name.startsWith(SNAPSHOT_DIR_NAME));
+    if (versionFiles == null || versionFiles.length == 0) {
+      return snapShotIndex;
+    }
+    for (File file : versionFiles) {
+      snapShotIndex =
+          Long.max(
+              snapShotIndex, Long.parseLong(file.getName().replaceAll(".*[^\\d](?=(\\d+))", "")));
```

Comment:
We may compile this  Pattern once as a class field?

Reply:
Already changed.